testsuite/integration/microprofile-tck/telemetry/src/test/java/io/smallrye/opentelemetry/arquillian/DeploymentProcessor.java

```diff
@@ -30,7 +31,9 @@ public void process(Archive<?> archive, TestClass testClass) {
             extensionsJar.addAsServiceProvider(Providers.class, ExceptionMapper.class);
 
             WebArchive war = WebArchive.class.cast(archive);
-            war.addAsLibraries(extensionsJar);
+            war
+                    .addAsLibraries(extensionsJar)
+                    .addAsWebInfResource(new StringAsset("<beans bean-discovery-mode=\"all\"/>"), "beans.xml");
```

Comment:
Not saying a -1 to this, but what was the reason for this being required? It feels like a bit of a bug, but I could be wrong.

Reply:
Without it, the injection is not performed in three tests for reasons I can't explain. I should, perhaps, see what changed in those tests.